File: other/tokilearning/Latihan/bilanganagakprima.cpp
#include <stdio.h>
#include <math.h>

using namespace std;

int main(){
    
    int c,t,n;
    
    scanf("%d",&t);
    
    for (int i=0; i<t; i++){
        scanf("%d",&n);
        if (n<8){
                 printf("YA\n");
                 continue;
                 }
        c=0;
        for (int j=2; j<=floor(sqrt(n)); j++)
            if (n%j==0) c++;
        if (c<=1) printf("YA\n");
           else printf("TIDAK\n");
           }
    return 0;
}

File: other/tokilearning/Latihan/bilanganagakprima.pas
var
  t,kasus,n,i,c:longint;

begin
  readln(t);
  for kasus:=1 to t do
    begin
      readln(n);
      if n<5 then
        begin
          writeln('YA');
          continue;
        end;
      c:=0;
      for i:=2 to trunc(sqrt(n)) do
        if n mod i=0 then
          begin
            inc(c);
            if c=2 then break;
          end;
      if c=2 then writeln('TIDAK')
        else writeln('YA');
    end;
end.
